WeakAuras/WeakAuras.lua

```diff
@@ -4694,44 +4730,38 @@ function WeakAuras.HandleGlowAction(actions, region)
       glow_frame = WeakAuras.GetUnitNameplate(region.state.unit)
     end
 
-    if glow_frame then
-      if not glow_frame.__WAGlowFrame then
-        glow_frame.__WAGlowFrame = CreateFrame("Frame", nil, glow_frame)
-        glow_frame.__WAGlowFrame:SetAllPoints(glow_frame)
-        glow_frame.__WAGlowFrame:SetSize(glow_frame:GetSize())
-      end
-      original_glow_frame = glow_frame
-      glow_frame = glow_frame.__WAGlowFrame
-    end
-
     if glow_frame then
       local id = region.id .. (region.cloneId or "")
       if actions.glow_action == "show" then
         actionGlowStart(actions, glow_frame, id)
-        region.active_glows = region.active_glows or {}
+        region.active_glows_hidefunc = region.active_glows_hidefunc or {}
         if actions.glow_frame_type == "UNITFRAME" then
           local glow_guid = UnitGUID(region.state.unit)
-          LGF.RegisterCallback("WeakAuras" .. id, "FRAME_GUID_UPDATE", function(event, frame, guid)
-            if frame ~= original_glow_frame and guid == glow_guid then
-              local new_glow_frame = WeakAuras.GetUnitFrame(region.state.unit)
-              if new_glow_frame and new_glow_frame ~= original_glow_frame then
-                actionGlowStop(actions, glow_frame, id)
-                if not new_glow_frame.__WAGlowFrame then
-                  new_glow_frame.__WAGlowFrame = CreateFrame("Frame", nil, new_glow_frame)
-                  new_glow_frame.__WAGlowFrame:SetAllPoints(new_glow_frame)
-                  new_glow_frame.__WAGlowFrame:SetSize(new_glow_frame:GetSize())
-                end
-                original_glow_frame = new_glow_frame
-                glow_frame = new_glow_frame.__WAGlowFrame
-                actionGlowStart(actions, glow_frame, id)
-              end
+          if glow_guid then
+            glow_frame_monitor = glow_frame_monitor or {}
+            glow_frame_monitor[glow_guid] = glow_frame_monitor[glow_guid] or {}
+            glow_frame_monitor[glow_guid][id] = {
+              region = region,
+              actions = actions,
+              frame = glow_frame
+            }
+            region.active_glows_hidefunc[glow_frame] = function()
+              actionGlowStop(actions, glow_frame, id)
+              glow_frame_monitor[glow_guid][id] = nil
+              -- TODO: purge glow_frame_monitor[glow_guid] if it's empty
```

Comment:
The way to check if a table is empty is via next(glow_frame_monitor[glow_guid]) == nil